proxy/http2/Http2Stream.cc

```diff
@@ -36,6 +36,15 @@
 
 ClassAllocator<Http2Stream> http2StreamAllocator("http2StreamAllocator");
 
+void
+Http2Stream::new_transaction()
+{
+  super::new_transaction();
+
+  ink_release_assert(this->current_reader != nullptr);
+  this->_http_sm_id = this->current_reader->sm_id;
```

Comment:
Wouldn't it be needed by H3?

Reply:
When H3 have Milestone/SlowLogs support.

Reply:
Then why not put it to the parent class? Are we not going to support it?

Reply:
The parent class is `ProxyTransaction` and it's also inherited by `Http1Transaction`. And `Http1Transaction` doesn't need this. Because Milestones for HTTP/1.1 is in HttpSM.
Milestones might be one thing of the refactoring SSN/TNX, but it's out of scope of this PR.

Reply:
> The parent class is ProxyTransaction and it's also inherited by Http1Transaction. And Http1Transaction doesn't need this. Because Milestones for HTTP/1.1 is in HttpSM.

This doesn't sounds like a good design. It's inconsistent. I don't think we should follow it, otherwise things get worse.

> Milestones might be one thing of the refactoring SSN/TNX, but it's out of scope of this PR.

I understand this PR is trying to fix a crash, but it seems like to me that you are unnecessarily adding a thing need to be refactored.

If you disagree, I can live with it but you need to find someone else to get an approval.

Reply:
I agree with Milestones, ProxyTransaction and HttpSM are not well designed.
But I don't want to do refactoring all of them here. Because 1) it might be huge changes 2) the refactoring `HttpSM` (#5488) is going on.

I'm fine with moving the assignment before the `update_read_request()` call instead of override `new_transaction`, if it looks simple.

Reply:
Who said you should refactor everything? 

Moving _http_sm_id to ProxyTransaction and initializing it in ProxyTransaction::new_transaction() are enough, no? Will it be a huge change? I think this is much simpler than overriding a method and it is also a right direction (no need to additional refactoring).

Reply:
@maskit how do you suggest the refactoring should be? as far as I understand H2 is using H1 and is getting H1 sm_id to link with H2 milestones. H2 and H1 State machines do not have a common base class. How do think @masaori335 should refactor the code?

Reply:
How is _httm_sm_id relevant in ProxyTransaction?

Reply:
Now, I understand your suggestion. But I'm not big fun of adding a member which is not used by all child classes.

If we add the member on the ProxyTransaction, it's not used by `Http1Transaction`. Also if we derive `Http(1|2|3)|ServerTransaction` from `ProxyTransaction`, the `_http_sm_id` will not used by them. These relations of classes are really unclear for now.

Reply:
> If we add the member on the ProxyTransaction, it's not used by Http1Transaction.

Currently true, and I think it should be changed later. Otherwise having it in a parent class doesn't make sense. Many things for H1 is currently too special because of historical reasons. I think new things should be designed more generically. Do you want to add another _http_sm_id to H3Transaction and override new_transaction again? I hope you don't forget to override it.

> Also if we derive Http(1|2|3)|ServerTransaction from ProxyTransaction, the _http_sm_id will not used by them. These relations of classes are really unclear for now.

I'd rename ProxyTransaction to ClientProxyTransaction and define a new ProxyTransaction that have common parts. I don't think _http_sm_id is the only member we don't need for server side.